src/Simulation/Simulators/SparseSimulator2/NativeWrappers.cs

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Common;
+
+namespace Microsoft.Quantum.Simulation.Simulators
+{
+    using QubitIdType = System.IntPtr;
+    using SimulatorIdType = System.UInt32;
+
+    public partial class SparseSimulator2
+    {
+        protected override void MCX(uint count, uint[] ctrls, uint qubit)
+        {
+            //MCXNative(this.Id, count, ctrls, qubit);
+
+            // private static extern void MCXNative(uint id, uint count, uint[] ctrls, uint qubit);
+            // private static extern void MCX_cpp(SimulatorIdType sim, int length, QubitIdType[] controls, QubitIdType target);
+            QubitIdType[] controls = new QubitIdType[ctrls.Length];
+            for(int i = 0; i < ctrls.Length; ++i)
+            {
+                controls[i] = (QubitIdType)(ctrls[i]);
+            }
```

Comment:
Using LINQ here would make this much more readable and less subject to errors in manual indexing.

```suggestion
            var controls = ctrls.Cast<QubitIdType>().ToArray();
```

Relevant extension method: https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.cast?view=net-6.0

Reply:
I like this mechanism very much, thank you!
FYI: I will specify the type (of `controls`) explicitly (`QubitIdType[] controls = `) rather than with `var`, since `var` makes `controls` _nullable_.